pentasharp/Controllers/BusinessBusCompanyController.cs

```diff
@@ -0,0 +1,113 @@
+﻿using AutoMapper;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using pentasharp.Data;
+using pentasharp.Interfaces;
+using pentasharp.Models.Enums;
+using pentasharp.Models.Utilities;
+using pentasharp.ViewModel.Bus;
+using WebApplication1.Filters;
+
+namespace WebApplication1.Controllers
+{
+    [Route("Business/BusCompany")]
+    [TypeFilter(typeof(BusinessOnlyFilter), Arguments = new object[] { "BusCompany" })]
+    public class BusinessBusCompanyController : Controller
+    {
+        private readonly IBusCompanyService _companyService;
+        private readonly IBusService _busService;
+        private readonly AppDbContext _context;
+        private readonly IHttpContextAccessor _httpContextAccessor;
+        private readonly IMapper _mapper;
+
+        public BusinessBusCompanyController(
+            IBusService busService,
+            IHttpContextAccessor httpContextAccessor,
+            AppDbContext context,
+            IMapper mapper,
+            IBusCompanyService companyService)
+        {
+            _busService = busService;
+            _httpContextAccessor = httpContextAccessor;
+            _context = context;
+            _mapper = mapper;
+            _companyService = companyService;
+        }
+
+        [HttpGet("GetCompany")]
+        public async Task<IActionResult> GetCompanyAsync()
+        {
+            var userId = GetUserId();
+            if (userId == null)
```

Comment:
as per other comments, move logic to the service of this endpoint and all others in this controller